plugin/src/main.rs

```diff
@@ -0,0 +1,94 @@
+use clap::Parser;
+use kube::Client;
+use kubectl_plugin::resources;
+use plugin::ExecuteOperation;
+use std::{env, ops::Deref};
+
+pub(crate) mod cli_utils;
+
+#[derive(Parser, Debug)]
+#[clap(name = utils::package_description!(), version = utils::version_info_str!())]
+#[group(skip)]
+struct CliArgs {
+    #[clap(subcommand)]
+    operations: cli_utils::Operations,
+
+    /// Kubernetes namespace of openebs service
+    #[clap(global = true, long, short = 'n')]
+    namespace: Option<String>,
+
+    #[clap(flatten)]
+    args: cli_utils::CliArgs,
+}
+
+impl CliArgs {
+    async fn args() -> Self {
+        let mut arg = CliArgs::parse();
+        if let Some(ns) = arg.namespace.clone() {
+            arg.args.args.namespace = ns;
+        } else {
+            let client = Client::try_default().await.expect("Client init failed");
+            let ns = client.default_namespace().to_string();
+            arg.args.args.namespace = ns;
+        }
+        arg
+    }
+}
+
+impl Deref for CliArgs {
+    type Target = plugin::CliArgs;
+
+    fn deref(&self) -> &Self::Target {
+        &self.args
+    }
+}
+
+#[tokio::main]
+async fn main() {
+    let cli_args = CliArgs::args().await;
+    let _tracer_flusher = cli_args.init_tracing();
+    if let Err(error) = cli_args.execute().await {
+        let mut exit_code = 1;
+        match error {
+            cli_utils::Error::Mayastor(variants) => match variants {
+                resources::Error::RestPlugin(error) => eprintln!("{error}"),
+                resources::Error::RestClient(error) => {
+                    eprintln!("Failed to initialise the REST client. Error {error}")
+                }
+                resources::Error::Upgrade(error) => {
+                    eprintln!("{error}");
+                    exit_code = error.into();
+                }
+                resources::Error::Generic(error) => eprintln!("{error}"),
+            },
+            cli_utils::Error::LocalpvLvm(error) => match error {
+                cli_utils::localpvlvm::Error::Generic(error) => eprint!("{error}"),
+                cli_utils::localpvlvm::Error::Kube(error) => eprint!("{error}"),
+            },
+            cli_utils::Error::LocalpvZfs(error) => match error {
+                cli_utils::localpvzfs::Error::Generic(error) => eprintln!("{error}"),
+            },
+            cli_utils::Error::Hostpath(error) => match error {
+                cli_utils::localpvhostpath::Error::Generic(error) => eprintln!("{error}"),
+            },
+            cli_utils::Error::ClusterInfo(error) => match error {
+                cli_utils::clusterinfo::Error::Generic(error) => eprintln!("{error}"),
+            },
+            cli_utils::Error::Generic(error) => eprintln!("{error}"),
```

Comment:
Do we need to match each specifically?
```suggestion
            error => eprint!("{error}"),
```

Reply:
hmm that's telling me that we don't have the errors defined properly, shouldn't see duplicate messages when printing the top error

Reply:
What do you mean by duplicate message? localpv will have similar type of errors. It may diverge if we add set capabilities.

Reply:
We have added snafu to error variants